src/nvim/ui.c

```diff
@@ -171,6 +171,11 @@ void ui_refresh(void)
     pum_external &= ui->pum_external;
   }
 
+  if (updating_screen) {
+    loop_schedule(&main_loop, event_create(1, ui_refresh_event, 0));
+    return;
+  }
```

Comment:
> if block could be moved before the declarations

Good point.  The current placement was due to me trying different things.  If it gets here, it usually happens a dozen or so times.  So, I was testing the width and height to double check that it wasn't from multiple SIGWINCH events.

> and loop_schedule(...) could be replaced with ui_schedule_refresh().

![](http://sae.tweek.us/media/emoticons/emot-doh.gif)